packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -352,8 +352,9 @@ export class GarbageCollector implements IGarbageCollector {
 	/**
 	 * API for ensuring the correct auto-recovery mitigations
 	 */
+	// eslint-disable-next-line unicorn/consistent-function-scoping -- Property is defined once; no need to extract inner lambda
```

Comment:
I agree with the disable. But purely out of curiosity I tried to think what would be the benefit of extracting the lambda. According the consistent-function-scoping docs, I think a bit of optimization since the function could be _defined_ once and optimized "a bit more statically" (so multiple instances of the parent `GarbageCollector` object could benefit from the optimization, maybe?) vs creating a new anonymous function like this does today? But the use case feels like it wouldn't benefit a lot from that, and the current approach probably can remove the function from memory after the first time it's called...? </speculation>

Reply:
You're probably right that extracting it could yield better static analysis results. I'll update this to be a TODO instead, so someone can look at it in the future.

Reply:
```suggestion
	// TODO: consider hoisting this to an outer scope as an optimization
	// eslint-disable-next-line unicorn/consistent-function-scoping
```